$MirOS: ports/www/firesomething/patches/patch-config_rules_mk,v 1.5.2.2 2008/11/23 14:37:52 bsiegert Exp $
$OpenBSD: patch-config_rules_mk,v 1.2 2004/12/27 21:21:39 wilfried Exp $
--- config/rules.mk.orig	Wed Jun 25 01:29:23 2008
+++ config/rules.mk	Fri Nov 21 19:56:22 2008
@@ -467,7 +467,7 @@ endif
 endif
 endif
 
-ifeq ($(OS_ARCH),FreeBSD)
+ifneq (,$(filter FreeBSD MidnightBSD MirBSD OpenBSD,$(OS_ARCH)))
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS += -Wl,-Bsymbolic
 endif
@@ -737,7 +737,7 @@ endif # DIST_INSTALL
 endif # LIBRARY
 ifdef SHARED_LIBRARY
 ifdef IS_COMPONENT
-	$(INSTALL) $(IFLAGS2) $(SHARED_LIBRARY) $(FINAL_TARGET)/components
+	$(INSTALL) $(IFLAGS1) $(SHARED_LIBRARY) $(FINAL_TARGET)/components
 	$(ELF_DYNSTR_GC) $(FINAL_TARGET)/components/$(SHARED_LIBRARY)
 ifdef BEOS_ADDON_WORKAROUND
 	( cd $(FINAL_TARGET)/components && $(CC) -nostart -o $(SHARED_LIBRARY).stub $(SHARED_LIBRARY) )
@@ -746,9 +746,9 @@ else # ! IS_COMPONENT
 ifneq (,$(filter OS2 WINNT WINCE,$(OS_ARCH)))
 	$(INSTALL) $(IFLAGS2) $(IMPORT_LIBRARY) $(DIST)/lib
 else
-	$(INSTALL) $(IFLAGS2) $(SHARED_LIBRARY) $(DIST)/lib
+	$(INSTALL) $(IFLAGS1) $(SHARED_LIBRARY) $(DIST)/lib
 endif
-	$(INSTALL) $(IFLAGS2) $(SHARED_LIBRARY) $(FINAL_TARGET)
+	$(INSTALL) $(IFLAGS1) $(SHARED_LIBRARY) $(FINAL_TARGET)
 ifdef BEOS_ADDON_WORKAROUND
 	( cd $(FINAL_TARGET) && $(CC) -nostart -o $(SHARED_LIBRARY).stub $(SHARED_LIBRARY) )
 endif
@@ -1012,7 +1012,7 @@ ifneq (,$(BUILD_STATIC_LIBS)$(FORCE_STAT
 LOBJS	+= $(SHARED_LIBRARY_LIBS)
 endif
 else
-ifneq (,$(filter OSF1 BSD_OS FreeBSD NetBSD OpenBSD SunOS Darwin,$(OS_ARCH)))
+ifneq (,$(filter BSD_OS Darwin FreeBSD MidnightBSD MirBSD NetBSD OSF1 OpenBSD SunOS,$(OS_ARCH)))
 CLEANUP1	:= | egrep -v '(________64ELEL_|__.SYMDEF)'
 CLEANUP2	:= rm -f ________64ELEL_ __.SYMDEF
 else
@@ -1180,7 +1180,6 @@ endif # Darwin
 else # os2 vacpp
 	$(MKSHLIB) -O:$@ -DLL -INC:_dllentry $(LDFLAGS) $(OBJS) $(LOBJS) $(EXTRA_DSO_LDOPTS) $(OS_LIBS) $(EXTRA_LIBS) $(DEF_FILE)
 endif # !os2 vacpp
-	chmod +x $@
 ifdef ENABLE_STRIP
 	$(STRIP) $@
 endif
